packages/backend-platform/src/WebsocketService.ts

```diff
@@ -0,0 +1,1348 @@
+  timeout?: number;
+
+  /** Initial reconnection delay in milliseconds (default: 500) */
+  reconnectDelay?: number;
+
+  /** Maximum reconnection delay in milliseconds (default: 5000) */
+  maxReconnectDelay?: number;
+
+  /** Request timeout in milliseconds (default: 30000) */
+  requestTimeout?: number;
+};
+
+/**
+ * Client Request message
+ * Used when client sends a request to the server
+ */
+export type ClientRequestMessage = {
+  event: string;
+  data: {
+    requestId: string;
+    channels?: string[];
+    [key: string]: unknown;
+  };
+};
+
+/**
+ * Server Response message
+ * Used when server responds to a client request
+ */
+export type ServerResponseMessage = {
+  event: string;
+  data: {
+    requestId: string;
+    subscriptionId?: string;
+    succeeded?: string[];
+    failed?: string[];
+    [key: string]: unknown;
+  };
+};
+
+/**
+ * Server Notification message
+ * Used when server sends unsolicited data to client
+ * subscriptionId is optional for system-wide notifications
+ */
+export type ServerNotificationMessage = {
+  event: string;
+  subscriptionId?: string;
+  channel: string;
+  data: Record<string, unknown>;
+};
+
+/**
+ * Union type for all WebSocket messages
+ */
+export type WebSocketMessage =
+  | ClientRequestMessage
+  | ServerResponseMessage
+  | ServerNotificationMessage;
+
+/**
+ * Internal subscription storage with full details including callback
+ */
+export type InternalSubscription = {
+  /** Channel names for this subscription */
+  channels: string[];
+  /** Callback function for handling notifications */
+  callback: (notification: ServerNotificationMessage) => void;
+  /** Function to unsubscribe and clean up */
+  unsubscribe: () => Promise<void>;
+};
+
+
+/**
+ * Channel-based callback configuration
+ */
+export type ChannelCallback = {
+  /** Channel name to match (also serves as the unique identifier) */
+  channelName: string;
+  /** Callback function */
+  callback: (notification: ServerNotificationMessage) => void;
+};
+
+
+/**
+ * External subscription info with subscription ID (for API responses)
+ */
+export type SubscriptionInfo = {
+  /** The subscription ID from the server */
+  subscriptionId: string;
+  /** Channel names for this subscription */
+  channels: string[];
+  /** Function to unsubscribe and clean up */
+  unsubscribe: () => Promise<void>;
+};
+
+/**
+ * Public WebSocket subscription object returned by the subscribe method
+ */
+export type WebSocketSubscription = {
+  /** The subscription ID from the server */
+  subscriptionId: string;
+  /** Function to unsubscribe and clean up */
+  unsubscribe: () => Promise<void>;
+};
+
+/**
+ * WebSocket connection info
+ */
+export type WebSocketConnectionInfo = {
+  state: WebSocketState;
+  url: string;
+  reconnectAttempts: number;
+  lastError?: string;
+  connectedAt?: number;
+};
+
+// Action types for the messaging system
+export type WebSocketServiceInitAction = {
```

Comment:
See note in `AccountActivityService` for DRYing up these types.

Reply:
Done